src/main/groovy/io/seqera/wave/service/builder/ContainerBuildServiceImpl.groovy

```diff
@@ -161,6 +161,44 @@ class ContainerBuildServiceImpl implements ContainerBuildService {
         }
     }
 
+    //this method adds labels in singularity container
+    protected static String addLabels(String containerFile, BuildRequest req){
```

Comment:
I would change the signature of the method to `String containerFile, Map<String, String> labels` , it is clearer.  
Then the 3 instances of `req.labels` in the body of the method also need to change to `labels`.

Reply:
I need to know the type of build also, so Its better to have req object